core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -426,25 +426,25 @@ class HttpGoogleServicesDAO(
     }
   }
 
-  override def getBucket(bucketName: String, userProject: Option[GoogleProjectId])(implicit executionContext: ExecutionContext): Future[Option[Bucket]] = {
+  override def getBucket(bucketName: String, userProject: Option[GoogleProjectId])(implicit executionContext: ExecutionContext): Future[Either[String, Bucket]] = {
     implicit val service = GoogleInstrumentedService.Storage
-    retryWithRecoverWhen500orGoogleError(() => {
+    retryWithRecoverWhen500orGoogleError[Either[String, Bucket]](() => {
       val getter = getStorage(getBucketServiceAccountCredential).buckets().get(bucketName)
       userProject.map( p => getter.setUserProject(p.value) )
 
-      Option(executeGoogleRequest(getter))
+      Right(executeGoogleRequest(getter))
```

Comment:
So the Right returns the Bucket variable? But what if it's just a String (doesn't Either main it can be either a String or a Bucket?)?

Reply:
Oh, I guess this is setting the return value and saying it is a Bucket?

Reply:
`Either` is often used in a scala convention to mean "bad thing or good thing", with `Right` being the "right" thing and `Left` being the "wrong" thing. In this case, the Bucket is the "right" think and the `String` is the wrong thing - ie an error message.

Reply:
Got it, thanks!